### PR TITLE
Do not transform when checking + check code reorg

### PR DESCRIPTION
**Pull request by elazarg**
* Avoiding ebpf_transform on check should improve performance
* Unify verifier calls to a single function with options parameter, returs ebpf_verifier_stats_t 
--  Use unique_ptr for cfg_t input, since it is moved from
-- Out parameter for invariants that should be returned
* Add the assume commands that triggered unreachability
* More constness on ebpf_checker: should not mutate the invariant
* No reachability notes on assume-assert
* Use dedicated exceptions for input errors
* Remove line_info from raw_program

## Release Notes

- **New Features**
  - Introduced custom exception types for improved error handling in control flow and ELF file processing.
  - Added a new `analyze_params_t` structure to streamline eBPF analysis parameters.
  - Enhanced the `ebpf_checker` class to allow more modular requirement handling.
  - Added new test cases for various jump conditions, loops, and bitwise operations.

- **Bug Fixes**
  - Improved error reporting for map descriptor retrieval failures.

- **Tests**
  - Expanded test coverage with new cases for conditional jumps, loops, and bitwise operations.
  - Updated existing tests to clarify conditions leading to unreachable code and ensure accurate postconditions.

- **Documentation**
  - Improved clarity and organization of output messages in test cases.

### CODE REVIEW

**Comment by elazarg** [comment]
(I don't think I understand the valgrind failure)

**Comment by Alan-Jowett** [comment]
> (I don't think I understand the valgrind failure)

Can you point me to the valgrind failure?

**Comment by elazarg** [comment]
> Can you point me to the valgrind failure?

I returned pass instead of !pass. Coderabbit caught it (which is kinda impressive)

File: src/crab_verifier.hpp [hunk 1]
```diff
@@ -7,18 +7,21 @@
...
+    // if exists, it will be moved from
+    std::unique_ptr<cfg_t> cfg{};
+    // invariants will be added to labels that appear as keys in out_invariants
+    const string_invariant* entry_invariant{};
+    string_invariant_map* out_invariants{};
+    const program_info* info{};
+    ebpf_verifier_options_t* options{};
+};
+
+ebpf_verifier_stats_t analyze_and_report(const analyze_params_t& params = {});
```
[the remark this answers is not in the document]
Comment (elazarg) [reply src/crab_verifier.hpp:24]: Where are the uninitialized pointers? The fields of the struct have default initilization.

File: src/crab_verifier.hpp [hunk 2]
```diff
@@ -7,18 +7,21 @@
...
+    std::ostream* os{&std::cout};
+    const InstructionSeq* prog{};
+    // if exists, it will be moved from
+    std::unique_ptr<cfg_t> cfg{};
+    // invariants will be added to labels that appear as keys in out_invariants
+    const string_invariant* entry_invariant{};
+    string_invariant_map* out_invariants{};
+    const program_info* info{};
+    ebpf_verifier_options_t* options{};
+};
```
[the remark this answers is not in the document]
Comment (elazarg) [reply src/crab_verifier.hpp:22]: None is mandatory.

**Comment by elazarg** [comment]
Superceded by #796

**Pull request closed without merging**